internal/services/send_invitation_message_test.go

```diff
@@ -0,0 +1,160 @@
+package services
+
+import (
+	"context"
+	"errors"
+	"net/url"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"github.com/stellar/stellar-disbursement-platform-backend/db"
+	"github.com/stellar/stellar-disbursement-platform-backend/db/dbtest"
+	"github.com/stellar/stellar-disbursement-platform-backend/internal/data"
+	"github.com/stellar/stellar-disbursement-platform-backend/internal/htmltemplate"
+	"github.com/stellar/stellar-disbursement-platform-backend/internal/message"
+)
+
+func TestSendInvitationMessageOptions_Validate(t *testing.T) {
+	testCases := []struct {
+		name    string
+		options SendInvitationMessageOptions
+		errStr  string
+	}{
+		{
+			name:    "missing first name",
+			options: SendInvitationMessageOptions{},
+			errStr:  "first name is required",
+		},
+		{
+			name:    "missing email",
+			options: SendInvitationMessageOptions{FirstName: "foobar"},
+			errStr:  "email is required",
+		},
+		{
+			name:    "missing role",
+			options: SendInvitationMessageOptions{FirstName: "foobar", Email: "foo@bar.com"},
+			errStr:  "role is required",
+		},
+		{
+			name:    "missing ui base URL",
+			options: SendInvitationMessageOptions{FirstName: "foobar", Email: "foo@bar.com", Role: "owner"},
+			errStr:  "UI base URL is required",
+		},
+		{
+			name:    "invalid ui base URL",
+			options: SendInvitationMessageOptions{FirstName: "foobar", Email: "foo@bar.com", Role: "owner", UIBaseURL: "%invalid$%"},
+			errStr:  `UI base URL is not a valid URL: parse "%invalid$%": invalid URL escape "%in"`,
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			err := tc.options.Validate()
+			if tc.errStr != "" {
+				assert.EqualError(t, err, tc.errStr)
+			} else {
+				assert.NoError(t, err)
+			}
+		})
+	}
+}
+
+func Test_SendInvitationMessage(t *testing.T) {
+	dbt := dbtest.Open(t)
+	defer dbt.Close()
+
+	dbConnectionPool, err := db.OpenDBConnectionPool(dbt.DSN)
+	require.NoError(t, err)
+	defer dbConnectionPool.Close()
+
+	messengerClientMock := message.MessengerClientMock{}
```

Comment:
You should be creating one new instance of this mock per sub-test, otherwise the sub-tests will influence each other and you won't be able to debug which one is failing or succeeding.